### PR TITLE
JIT: ARR_LENGTH(new T[CNS]) --> CNS

### PR DESCRIPTION
**Pull request by EgorBo**
Quick 5 lines-of-code change to enable `ARR_LENGTH(JitNewArr)` constant folding in VN:

```csharp
int Test()
{
    var a = new int[100];
    return a.Length;
}
```
Current codegen:
```asm
; Method Proga:Test():int:this
       sub      rsp, 40
       mov      rcx, 0xD1FFAB1E      ; int[]
       mov      edx, 100
       call     CORINFO_HELP_NEWARR_1_VC
       mov      eax, dword ptr [rax+08H]
       add      rsp, 40
       ret      
; Total bytes of code: 32
```

New codegen:
```asm
; Method Proga:Test():int:this
       mov      eax, 100
       ret      
; Total bytes of code: 6
```

I recommend enabling "hide whitespaces" for code review since most of the changes are just extra { } block

### CODE REVIEW

**Comment by EgorBo** [comment]
@dotnet/jit-contrib PTAL, simple change. [diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=255349&view=ms.vss-build-web.run-extensions-tab)

**Review by TIHan — APPROVED** [review]

File: src/coreclr/jit/valuenum.cpp [hunk 1]
```diff
@@ -2401,37 +2401,51 @@ ValueNum ValueNumStore::VNForFunc(var_types typ, VNFunc func, ValueNum arg0VN)
...
+                                    {
+                                        resultVN = VNForIntCon(len);
+                                    }
                                 }
                             }
                         }
                     }
                 }
+                // Case 3: ARR_LENGTH(new T[CNS])
+                else if ((funcApp.m_func == VNF_JitNewArr) && IsVNConstant(funcApp.m_args[1]))
```
[the remark this answers is not in the document]
Comment (EgorBo) [reply src/coreclr/jit/valuenum.cpp:2439]: yeah, my first impl used `GetNewArrSize` but for TP reasons I inlined it here, forgot about R2R

[the remark this answers is not in the document]
Comment (EgorBo) [reply src/coreclr/jit/valuenum.cpp]: Simplified the impl, GetVNFunc is still here, if TP report is fine with that then I don't think I need to complicate it more.

**Comment by EgorBo** [comment]
[New diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=255896&view=ms.vss-build-web.run-extensions-tab), a tiny bit better than previously

**Pull request merged**